vcr/credential/validator.go

```diff
@@ -176,16 +176,19 @@ func (d nutsAuthorizationCredentialValidator) Validate(credential vc.VerifiableC
 	if len(strings.TrimSpace(cs.PurposeOfUse)) == 0 {
 		return failure("'credentialSubject.PurposeOfUse' is nil")
 	}
-	switch cs.LegalBase.ConsentType {
-	case "implied":
-		// no additional requirements
-		break
-	case "explicit":
-		if cs.Subject == nil || len(strings.TrimSpace(*cs.Subject)) == 0 {
-			return failure("'credentialSubject.Subject' is required when consentType is 'explicit'")
+
+	if credential.ContainsContext(NutsV2ContextURI) {
+		switch cs.LegalBase.ConsentType {
+		case "implied":
+			// no additional requirements
+			break
+		case "explicit":
+			if cs.Subject == nil || len(strings.TrimSpace(*cs.Subject)) == 0 {
```

Comment:
Instead of `len(…) == 0` a `strings.TrimSpace(*cs.Subject) != ""` would read nicely here. The Go compiler is optimised for this scenario.

Reply:
Assuming _The Internet_ is right; I had to do some digging for proof, but it indeed seems to be faster (or at least lead to less operations); https://stackoverflow.com/a/71202228/1651690

Reply:
😆 👍  We just have to know.

Reading the output from `go build -gcflags=-S` on a regular basis comes highly recommend b.t.w.. The compiler almost never does what I expect it to do.